### PR TITLE
move sources site to mirrors

### PR DESCRIPTION
**Pull request by classabbyamp**
- ansible/host_vars/{a-fra-de,b-hel-fi,d-hel-fi}: add dist_sources nomad vol
- services/nomad/mirror/shadow-rsyncd: serve sources on rsync
- services/nomad/mirror/sync: sync sources to mirrors
- services/nomad/infrastructure/nginx-mirror: serve sources
- terraform/do/dns: sources uses omniproxy

### CODE REVIEW

File: services/nomad/mirror/sync.nomad [hunk 1]
```diff
@@ -55,5 +62,42 @@ EOF
         destination = "/mirror"
       }
     }
+
+    task "sources-rsync" {
+      leader = true
```
Comment (classabbyamp) [inline services/nomad/mirror/sync.nomad:67]: this might not be right

Comment (the-maldridge) [reply services/nomad/mirror/sync.nomad:67]: In this case you probably don't need to set a leader at all.  A leader is set when a clear relationship exists between one task and a group of dependent workers, which is not the case here.

File: services/nomad/infrastructure/nginx-mirror.nomad [hunk 2]
```diff
@@ -14,6 +14,12 @@ job "nginx" {
       read_only = true
     }
 
+    volume "dist-sources" {
```
Comment (classabbyamp) [inline services/nomad/infrastructure/nginx-mirror.nomad:17]: can probably be dynamic

File: services/nomad/infrastructure/nginx-mirror.nomad [hunk 3]
```diff
@@ -28,7 +34,12 @@ job "nginx" {
 
       volume_mount {
         volume = "dist-mirror"
-        destination = "/srv/www"
+        destination = "/srv/www/mirror"
+      }
+
+      volume_mount {
+        volume = "dist-mirror"
```
Comment (classabbyamp) [inline services/nomad/infrastructure/nginx-mirror.nomad:41]: dist-sources

File: services/nomad/infrastructure/nginx-mirror.nomad [hunk 4]
```diff
@@ -26,9 +31,13 @@ job "nginx" {
...
       }
 
-      volume_mount {
-        volume = "dist-mirror"
-        destination = "/srv/www"
+      dynamic "volume_mount" {
+        for_each = [ "mirror", "sources", ]
+
+        content {
+          volume = "dist-${volume.value}"
```
Comment (the-maldridge) [inline services/nomad/infrastructure/nginx-mirror.nomad:38]: s/volume/volume_mount/

**Pull request closed without merging**